src/include/txn_inline.h

```diff
@@ -793,11 +793,11 @@ __txn_visible_id(WT_SESSION_IMPL *session, uint64_t id)
 }
 
 /*
- * __wt_txn_visible --
- *     Can the current transaction see the given ID / timestamp?
+ * __wt_txn_timestamp_visible --
+ *     Can the current transaction can see the given timestamp?
  */
 static inline bool
-__wt_txn_visible(
+__wt_txn_timestamp_visible(
   WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t timestamp, wt_timestamp_t durable_timestamp)
```

Comment:
Ideally, I wouldn't be passing ```id``` to the timestamp-only visibility check. ```__txn_visible_id()``` already checks for the current transaction's id. We can potentially remove the id check here. You will have to modify ```__wt_txn_snap_min_visible()``` to adjust to the current id check removal.

Reply:
No, the check in the function `__txn_visible_id()` is to ensure that the transaction id is visible or not. The following check is to ensure whether we need to perform the timestamp check or not.

If I need to remove that check in the `timestamp_visible` function, I have to copy a similar check in both functions.

Reply:
I meant to move the id check into id check functions and away from timestamps, even if we have to move the check at couple of places:

```    if (F_ISSET(session->txn, WT_TXN_HAS_ID) && id == session->txn->id)
        return (true)```

I think this is more aesthetic than functional reasons, so feel free to ignore this comment.

Reply:
Understood. Fixed it by moving the same transaction id check to the caller functions.